diff -ur mate-file-manager-1.4.0-patched/eel/eel-glib-extensions.c work/src/mate-file-manager-1.4.0/eel/eel-glib-extensions.c
--- mate-file-manager-1.4.0-patched/eel/eel-glib-extensions.c	2012-07-05 17:58:54.000000000 +0000
+++ work/src/mate-file-manager-1.4.0/eel/eel-glib-extensions.c	2013-05-28 11:50:31.002053553 +0000
@@ -967,14 +967,14 @@
 gboolean
 eel_get_filename_charset (const gchar **filename_charset)
 {
-    static GStaticPrivate cache_private = G_STATIC_PRIVATE_INIT;
-    EelFilenameCharsetCache *cache = g_static_private_get (&cache_private);
+    static GPrivate cache_private = G_PRIVATE_INIT(g_free);
+    EelFilenameCharsetCache *cache = g_private_get (&cache_private);
     const gchar *charset;
 
     if (!cache)
     {
         cache = g_new0 (EelFilenameCharsetCache, 1);
-        g_static_private_set (&cache_private, cache, filename_charset_cache_free);
+        g_private_set (&cache_private, cache);
     }
 
     g_get_charset (&charset);
diff -ur mate-file-manager-1.4.0-patched/libcaja-private/caja-debug-log.c work/src/mate-file-manager-1.4.0/libcaja-private/caja-debug-log.c
--- mate-file-manager-1.4.0-patched/libcaja-private/caja-debug-log.c	2012-07-05 17:58:53.000000000 +0000
+++ work/src/mate-file-manager-1.4.0/libcaja-private/caja-debug-log.c	2013-05-28 11:53:21.055063050 +0000
@@ -39,7 +39,7 @@
 
 #define MAX_URI_COUNT 20
 
-static GStaticMutex log_mutex = G_STATIC_MUTEX_INIT;
+G_LOCK_DEFINE_STATIC(log_mutex);
 
 static GHashTable *domains_hash;
 static char **ring_buffer;
@@ -53,13 +53,13 @@
 static void
 lock (void)
 {
-    g_static_mutex_lock (&log_mutex);
+    G_LOCK(log_mutex);
 }
 
 static void
 unlock (void)
 {
-    g_static_mutex_unlock (&log_mutex);
+    G_UNLOCK(log_mutex);
 }
 
 void
