diff -ur libmatecomponent-1.2.1/matecomponent-activation/matecomponent-activation-client.h libmatecomponent-1.2.1-topatch//matecomponent-activation/matecomponent-activation-client.h
--- libmatecomponent-1.2.1/matecomponent-activation/matecomponent-activation-client.h	2012-03-27 18:52:14.000000000 +0000
+++ libmatecomponent-1.2.1-topatch//matecomponent-activation/matecomponent-activation-client.h	2012-07-09 15:37:36.000000000 +0000
@@ -2,6 +2,13 @@
 #define MATECOMPONENT_ACTIVATION_CLIENT_H
 
 #include <glib.h>
+
+/* [FIXME: redeclared to get this to compile, but officially it's
+   deprecated. I'll let the mate team take care of this. */
+#ifdef G_DISABLE_DEPRECATED
+typedef void (*GVoidFunc) (void);
+#endif 
+
 #include <matecomponent-activation-private.h>
 
 void matecomponent_activation_release_corba_client (void);
